### PR TITLE
init for custom blacklist

### PR DESCRIPTION
**Pull request by 6fears7**
Still need to address resizble width of the remix container. If users keep typing they will eventually be blocked from adding more mods due to width.

### CODE REVIEW

File: Online/RainMeadowModManager.cs [hunk 1]
```diff
@@ -8,54 +8,49 @@ namespace RainMeadow
     public class RainMeadowModManager
     {
         public static string[] GetActiveMods()
-        { 
+        {
             return ModManager.ActiveMods
                 .Where(mod => Directory.Exists(Path.Combine(mod.path, "modify", "world"))
-                    || highImpactMods.Contains(mod.id)
-                    || cheatMods.Contains(mod.id))
+                    || blacklistMods.Contains(mod.id))
```
Comment (youbitchoc) [inline Online/RainMeadowModManager.cs:14]: this won't work

you have to sync the banlist and the require-list separately

File: Online/RainMeadowModManager.cs [hunk 2]
```diff
@@ -8,54 +8,49 @@ namespace RainMeadow
...
+            return mods.Intersect(blacklistMods).ToArray();
+
+
+
         }
         internal static void CheckMods(string[] lobbyMods, string[] localMods)
         {
-            var lobbyCheatMods = GetCheatMods(lobbyMods);
-            var localCheatMods = GetCheatMods(localMods);
+            var lobbyCheatMods = GetBlackList(lobbyMods);
```
Comment (youbitchoc) [inline Online/RainMeadowModManager.cs:52]: this logic will have to be changed too, also rename the vars to make more sense

File: Online/RainMeadowModManager.cs (hunk 1 above)
Comment (youbitchoc) [reply Online/RainMeadowModManager.cs:14]: handle overlaps by removing from banlist?

**Review by youbitchoc — CHANGES_REQUESTED** [review]
logic won't work as-is, need separate banlist and require-list

File: Online/RainMeadowModManager.cs (hunk 1 above)
Comment (6fears7) [reply Online/RainMeadowModManager.cs:14]: Tested by having host type in "rwremix" into blacklist. Client had remix enabled and was prevented from joining. Not sure what I'm missing from what you're saying.

File: Online/RainMeadowModManager.cs (hunk 2 above)
Comment (6fears7) [reply Online/RainMeadowModManager.cs:52]: thanks

File: Online/RainMeadowModManager.cs (hunk 1 above)
Comment (youbitchoc) [reply Online/RainMeadowModManager.cs:14]: the banlist is for mods that clients should disable
the requirelist is for mods that clients should enable

these two should not be lumped together

Comment (youbitchoc) [reply Online/RainMeadowModManager.cs:14]: yes, and this will require redoing a lot of logic

Comment (youbitchoc) [reply Online/RainMeadowModManager.cs:14]: > require (beyond "modify", "region")

world and enum-changing ones, order matters

**Pull request closed without merging**